packages/staking/solana/lib/rewards.ts

```diff
@@ -207,3 +210,7 @@ export const calcRewards = (
 
   return rewardEntryAccumulator.getClaimableAmount();
 };
+
+export const calculateRewardAmount = (rewardRate: number, rewardTokenDecimals: number, stakeTokenDecimals: number) => {
+  return getBN(rewardRate, rewardTokenDecimals + (REWARD_AMOUNT_PRECISION_FACTOR - stakeTokenDecimals));
```

Comment:
I'm completely unsure that getBN does work with negative numbers, Have you checked it?